crowbar_framework/app/models/deployer_service.rb

```diff
@@ -99,9 +101,11 @@ def transition(inst, name, state)
         break if done
       end
 
-      os_map = role.default_attributes["deployer"]["os_map"]
-      node.crowbar["crowbar"]["hardware"]["os"] = os_map[0]["install_os"]
-      node.save
+      unless node.crowbar["crowbar"]["hardware"].nil?
```

Comment:
It's `node.crowbar["crowbar"]["hardware"]` that is supposed to be filled on lines 94-96, so I focused on this check